lib/blocks.php

```diff
@@ -172,6 +172,40 @@ function gutenberg_reregister_core_block_types() {
 
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Adds the defer loading strategy to all registered blocks.
+ *
+ * This function would not be part of core merge. Instead, the register_block_script_handle() function would be patched
+ * as follows.
+ *
+ * ```
+ * --- a/wp-includes/blocks.php
+ * +++ b/wp-includes/blocks.php
+ * @ @ -153,7 +153,8 @ @ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
+ *                 $script_handle,
+ *                 $script_uri,
+ *                 $script_dependencies,
+ * -           isset( $script_asset['version'] ) ? $script_asset['version'] : false
+ * +         isset( $script_asset['version'] ) ? $script_asset['version'] : false,
+ * +         array( 'strategy' => 'defer' )
+ *         );
+ *         if ( ! $result ) {
+ *                 return false;
+ * ```
+ *
+ * @see register_block_script_handle()
+ */
+function gutenberg_defer_block_view_scripts() {
+	$block_types = WP_Block_Type_Registry::get_instance()->get_all_registered();
+	foreach ( $block_types as $block_type ) {
+		foreach ( $block_type->view_script_handles as $view_script_handle ) {
+			wp_script_add_data( $view_script_handle, 'strategy', 'defer' );
```

Comment:
@westonruter In your comments you mention that you only want to target core blocks in your initial approach. However, this here will affect any block that has been registered. Regardless of whether it is a core block, or a custom one from a 3rd party. Is this intended? 

(I'm definitely looking forward to 3rd party view scripts no longer being blocking in the head. But want to make sure we are not causing unexpected side effects here :) )

Reply:
@fabiankaegy Sorry, in which comment did I say that? Initially it was limited to core blocks, but eventually the scope applied to all blocks. The reason was discovered in https://github.com/WordPress/gutenberg/pull/52536#issuecomment-1642631741 where it turns out that in classic themes all block view scripts get printed in the footer anyway, so it means that we can safely add `defer` to all blocks, whether in core or in 3rd party.

Reply:
Ahh in that case I misread something and didn’t have the full context. Thanks for sharing that 👍